### PR TITLE
WIP: Set up Course Discovery IDA

### PR DESCRIPTION
**Pull request by cpennington**
@clintonb @feanil: Preview!

### CODE REVIEW

File: playbooks/roles/course_discovery/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    USER: 'course_discovery001'
+    PASSWORD: 'password'
+    HOST: 'localhost'
+    PORT: '3306'
+    ATOMIC_REQUESTS: true
+    CONN_MAX_AGE: 60
+
+COURSE_DISCOVERY_VERSION: "master"
+COURSE_DISCOVERY_DJANGO_SETTINGS_MODULE: "course_discovery.settings.production"
+COURSE_DISCOVERY_URL_ROOT: 'http://localhost:9222'
```
Comment (clintonb) [inline playbooks/roles/course_discovery/defaults/main.yml:44]: Why this port? The next logical port number after programs (8003) is 8004. Not sure if we want to continue the sequencing.

**Comment by clintonb** [comment]
@cpennington there is an undocumented step to handle DNS setup for sandboxes.
1. Find the `Add DNS names for services` task in `/playbooks/roles/launch_ec2/tasks/main.yml`.
2. Add your service's name to the list at the end of the task. This will create a DNS entry of the format `<service-name>-<username>.sandbox.edx.org` (e.g. `ecommerce-clintonb.sandbox.edx.org`).

Updating documentation now.

**Comment by clintonb** [comment]
There are a couple settings missing from `ansible-provision.sh`. See improved steps at https://github.com/edx/configuration/wiki/Creating-Django-based-Independently-Deployable-Applications-(IDAs)#sandboxes.

If anyone has ideas for automating the addition of this data to the various files, I am open to ideas. This process is still too manual.

**Comment by clintonb** [comment]
Once the above comments are addressed, I am happy. Let me know when the caching, elasticsearch, and other changes need review.

File: playbooks/roles/course_discovery/defaults/main.yml (hunk 1 above)
Comment (cpennington) [inline playbooks/roles/course_discovery/defaults/main.yml:44]: 8004 is already taken.

Comment (cpennington) [inline playbooks/roles/course_discovery/defaults/main.yml:44]: At least in vagrant, 8004 is LMS for BokChoy

**Comment by cpennington** [comment]
Replaced by #2491

**Pull request closed without merging**